x-pack/plugins/security_solution/scripts/endpoint/common/fleet_server/fleet_server_services.ts

```diff
@@ -0,0 +1,673 @@
+import { resolve } from 'path';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  RETRYABLE_TRANSIENT_ERRORS,
+  retryOnError,
+} from '../../../../common/endpoint/data_loaders/utils';
+import { isServerlessKibanaFlavor } from '../stack_services';
+import type { FormattedAxiosError } from '../format_axios_error';
+import { catchAxiosErrorFormatAndThrow } from '../format_axios_error';
+import {
+  fetchFleetOutputs,
+  fetchFleetServerHostList,
+  fetchFleetServerUrl,
+  fetchIntegrationPolicyList,
+  generateFleetServiceToken,
+  getAgentVersionMatchingCurrentStack,
+  getFleetElasticsearchOutputHost,
+  waitForHostToEnroll,
+} from '../fleet_services';
+import { dump } from '../../endpoint_agent_runner/utils';
+import { getLocalhostRealIp } from '../localhost_services';
+import { isLocalhost } from '../is_localhost';
+
+export const FLEET_SERVER_CUSTOM_CONFIG = resolve(__dirname, './fleet_server.yml');
+
+interface StartedServer {
+  /** The type of virtualization used to start the server */
+  type: 'docker';
+  /** The ID of the server */
+  id: string;
+  /** The name of the server */
+  name: string;
+  /** The url (including port) to the server */
+  url: string;
+  /** Stop server */
+  stop: () => Promise<void>;
+  /** Any information about the server */
+  info?: string;
+}
+
+interface StartFleetServerOptions {
+  kbnClient: KbnClient;
+  logger: ToolingLog;
+  /** Policy ID that should be used to enroll the fleet-server agent */
+  policy?: string;
+  /** Agent version */
+  version?: string;
+  /** Will start fleet-server even if its detected that it is already running */
+  force?: boolean;
+  /** The port number that will be used by fleet-server to listen for requests */
+  port?: number;
+}
+
+interface StartedFleetServer extends StartedServer {
+  /** The policy id that the fleet-server agent is running with */
+  policyId: string;
+}
+
+export const startFleetServer = async ({
+  kbnClient,
+  logger,
+  policy,
+  version,
+  force = false,
+  port = 8220,
+}: StartFleetServerOptions): Promise<StartedFleetServer> => {
+  logger.info(`Starting Fleet Server and connecting it to Kibana`);
+
+  const response = await logger.indent(4, async () => {
+    const isServerless = await isServerlessKibanaFlavor(kbnClient);
+
+    let policyId = policy ?? '';
+
+    // Check if fleet already running if `force` is false
+    if (!force) {
+      const currentFleetServerUrl = await fetchFleetServerUrl(kbnClient);
+
+      if (currentFleetServerUrl && (await isFleetServerRunning(currentFleetServerUrl))) {
+        throw new Error(
+          `Fleet server is already configured for this instance of Kibana and available at: ${currentFleetServerUrl}.\n(Use 'force' option to bypass this error)`
+        );
+      }
+    }
+
+    // Only fetch/create a fleet-server policy
+    if (!policyId && !isServerless) {
+      policyId = await getOrCreateFleetServerAgentPolicyId(kbnClient, logger);
```

Comment:
I think we can set `policyId` the same way as `serviceToken` below, it might get more readable, what do you think?